add reference docs for configuration options

tang-mm opened the pull request:
Add more reference docs from [Goolge drive]( https://docs.google.com/document/d/1-lcFL2XAsgCYpjxwje4fiAdbJwkfDfAZJucBNJ6tRKQ/edit#heading=h.ec3a1r1axbo), including:
- Device connector configuration options
- Supported device connector types
- Testfligner agent configuration options
- Testfligner server configuration options (environment variables)

This PR resolves story: [CERTTF-226](https://warthogs.atlassian.net/browse/CERTTF-226) and [CERTTF-225](https://warthogs.atlassian.net/browse/CERTTF-225)

[CERTTF-226]: https://warthogs.atlassian.net/browse/CERTTF-226?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CERTTF-225]: https://warthogs.atlassian.net/browse/CERTTF-225?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

plars commented on docs/reference/testflinger-agent-conf.rst:
```diff
@@ -0,0 +1,78 @@
+      - rpi4
+      - rpi4-001
+  setup_command: echo Nothing needed for setup
+  provision_command: snappy-device-agent muxpi provision -c /path/to/default.yaml testflinger.json
```
Please replace snappy-device-agent commands everywhere with "testflinger-device-connector" (both will work for now, but we're trying to encourage migration to the new command name)

tang-mm replied:
Updated with a new commit `0440672`

plars reviewed (changes requested):
Looking good, a few small updates below

tang-mm commented:
@plars I updated the command and config names per your comments. Could you review it again?
@nancyc12 I added the new `firmware_update` phase (#114 )  into the example. Could you take a look?

nancyc12 commented on docs/reference/testflinger-agent-conf.rst:
```diff
@@ -0,0 +1,81 @@
+      - rpi4-001
+  setup_command: echo Nothing needed for setup
+  provision_command: testflinger-device-connector muxpi provision -c /path/to/default.yaml testflinger.json
+  firmware_update_command: echo Updating device firmware to the latest version
```
```suggestion
  firmware_update_command: testflinger-device-connector muxpi firmware_update -c /path/to/default.yaml testflinger.json
```

nancyc12 reviewed (changes requested):
@tang-mm thanks for making the changes for the firmware_update_command. I added a suggestion for firmware_update_command example.

nancyc12 reviewed (approved):
LGTM +1

plars reviewed (approved):
Sorry, didn't see the update here, +1 from me!

plars commented:
resolved small merge conflict with the wordlist
